client/extensions/woocommerce/state/sites/product-categories/selectors.js

```diff
@@ -176,3 +181,24 @@ export function getTotalProductCategories(
 	const categoryState = getRawCategoryState( state, siteId );
 	return ( categoryState.total && categoryState.total[ serializedQuery ] ) || 0;
 }
+
+/*
+ * Get the label for a given product, recursively including parent names.
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} categoryId ID of the starting category.
+ * @param {Number} [siteId] wpcom site id, if not provided, uses the selected site id.
+ * @return {String} Label of given category, with all parents included
+ */
+function getProductCategoryLabel( state, categoryId, siteId = getSelectedSiteId( state ) ) {
+	const categoryState = getRawCategoryState( state, siteId );
+	const categories = categoryState.items || {};
+	const category = categories[ categoryId ];
+	if ( ! category ) {
+		return '';
```

Comment:
If all categories haven't been loaded into the state tree, i.e. a site has an absurd number of categories, could this result in something like `grandparent category - - grandchild category` when the middle parent category is not in the state tree? Edge case but just 🤔

Reply:
👍 yeah i seem to recall that now you mention it - and i want to say the default page size is 100 which will likely get all categories on 99% of the sites in one call